docs(introduction): Add a narrative quick start

rpatterson opened the pull request:
I wrote this as I worked, with the help of more knowledgeable folks on Discord, to get up and running with the use case of feeding all announcements to Servarr.  I was confused about what the larger overview of autobrr is, how it's parts are intended to work together, and I couldn't find anything like that in the docs.  So I captured the bits I was confused about as I encountered them and wrote them up in narrative form here in the hopes it might help others with similar confusions.

rpatterson commented:
I can't make sense of what the failed checks mean let alone if they're really related to my changes.  If this is anything I should address, please provide some guidance.  Thanks!

s0up4200 commented:
> I can't make sense of what the failed checks mean let alone if they're really related to my changes. If this is anything I should address, please provide some guidance. Thanks!

Not able to view the deploy log right now, but I'm pretty sure it's caused by broken links.
Remove the `.` in all your links and it should be fine!

s0up4200 commented on docs/introduction.md:
```diff
@@ -85,3 +85,71 @@ Available download clients and actions
+
+Now you can monitor the logs for announcements from IRC, pushes to the Servarr instance,
+and details for what, if anything, Servarr did with the release:
+
```
I'd probably use fenced code blocks for the log output here.

```text
INFO Matched 'Foo Series S01E01 1080p WEB h264-BarGrp' (All) for foo-indexer
DEBUG release.store: &{ID:12 FilterStatus:FILTER_APPROVED Rejections:[] Indexer:foo-indexer FilterName:All Protocol:torrent Implementation:IRC Timestamp:1970-01-01 00:00:00.000000000 -0000 UTC m=+0000.000000000 GroupID: TorrentID:######### TorrentURL:https://www.example.com/Foo+Series+S01E01+1080p+WEB+h264-BarGrp.torrent TorrentTmpFile: TorrentDataRawBytes:[] TorrentHash: TorrentName:Foo Series S01E01 1080p WEB h264-BarGrp Size:0 Title:Foo Series Category:TV :: Episodes HD Categories:[] Season:1 Episode:1 Year:0 Resolution:1080p Source:WEB Codec:[H.264] Container: HDR:[] Audio:[] AudioChannels: Group:BarGrp Region: Language: Proper:false Repack:false Website: Artists: Type: LogScore:0 IsScene:false Origin: Tags:[] ReleaseTags: Freeleech:false FreeleechPercent:0 Bonus:[] Uploader:Anonymous PreTime: Other:[] RawCookie: AdditionalSizeCheckRequired:false FilterID:1 Filter:0x########## ActionStatus:[]}
DEBUG sonarr: release push rejected: Foo Series S01E01 1080p WEB h264-BarGrp, indexer foo-indexer to http://localhost:8989 reasons: '[Unknown Series]'
DEBUG release rejected: Unknown Series
```

rpatterson replied:
Apologies, I'm much more of a reStructuredText devotee than I am knowledgeable in markdown.  Can you elaborate/educate?

s0up4200 replied:
Like this: https://www.markdownguide.org/extended-syntax/#fenced-code-blocks

rpatterson replied:
Ah, will do.  Why do you prefer this so I can learn?

rpatterson replied:
Hmm, that renders all lines as one line when I use the CLI renderer locally:

    $ markdown --version

    This is Markdown, version 1.0.1.
    Copyright 2004 John Gruber
    http://daringfireball.net/projects/markdown/

LMK if you still prefer the fenced version and I'll push the change.

s0up4200 replied:
It should not render them as one line if you use 3 backticks before and after the first and last line.
One backtick will put all in one line though.

I guess its just more appealing visually in the document. It looks messy if the log output is just a regular paragraph. That's just my opinion though 😄

rpatterson replied:
I like that argument for the developer/writer experience of reading the souce.  I pushed the change.  This version of it still renders as one line in my Chrome from that version of the CLI markdown tool, just FYI.  Odd:

    <p><code>
    INFO Matched 'Foo Series S01E01 1080p WEB h264-BarGrp' (All) for foo-indexer
    DEBUG release.store: &amp;{ID:12 FilterStatus:FILTER_APPROVED Rejections:[] Indexer:foo-indexer FilterName:All Protocol:torrent Implementation:IRC Timestamp:1970-01-01 00:00:00.000000000 -0000 UTC m=+0000.000000000 GroupID: TorrentID:######### TorrentURL:https://www.example.com/Foo+Series+S01E01+1080p+WEB+h264-BarGrp.torrent TorrentTmpFile: TorrentDataRawBytes:[] TorrentHash: TorrentName:Foo Series S01E01 1080p WEB h264-BarGrp Size:0 Title:Foo Series Category:TV :: Episodes HD Categories:[] Season:1 Episode:1 Year:0 Resolution:1080p Source:WEB Codec:[H.264] Container: HDR:[] Audio:[] AudioChannels: Group:BarGrp Region: Language: Proper:false Repack:false Website: Artists: Type: LogScore:0 IsScene:false Origin: Tags:[] ReleaseTags: Freeleech:false FreeleechPercent:0 Bonus:[] Uploader:Anonymous PreTime: Other:[] RawCookie: AdditionalSizeCheckRequired:false FilterID:1 Filter:0x########## ActionStatus:[]}
    DEBUG sonarr: release push rejected: Foo Series S01E01 1080p WEB h264-BarGrp, indexer foo-indexer to http://localhost:8989 reasons: '[Unknown Series]'
    DEBUG release rejected: Unknown Series
    </code></p>

rpatterson commented:
> Not able to view the deploy log right now, but I'm pretty sure it's caused by broken links. Remove the `.` in all your links and it should be fine!

I removed that specific form of relative link, but it still fails.  Also, there are preexisting examples of that form of relative links in the docs that I haven't touched:

    git --no-pager grep -E -n --color=auto -i -e '\]\(\./' -- '*.md' '**.md'
    configuration/autobrr.md:90:- [Indexers <CgWebsite />](./indexers)
    configuration/autobrr.md:91:- [IRC <BsChatDotsFill />](./irc)
    configuration/autobrr.md:92:- [Download Clients <FaDownload />](./download-clients)
    configuration/autobrr.md:93:- [Torznab Feeds <MdRssFeed />](./feeds)
    configuration/indexers.md:32:Please check the [IRC](./irc) section for more details regarding NickServ, IRC keys, and grouping of nicks.

Anything else I can do?

gl0ryus commented on docs/introduction.md:
```diff
@@ -85,3 +85,72 @@ Available download clients and actions
+
+5. Add an [indexer](./configuration/indexers).
+
+6. Add a [Servarr instance](./configuration/download-clients#sonarr)
```
This should probably just be 

6. Add a [Download client](./configuration/download-clients)

rpatterson replied:
That change might better fit the terminology of autobrr, but it would be less clear for someone coming in for this use case.  IOW, if I neutered this of that specific use case it would become much longer and I wouldn't be able to write it.  But if someone else agrees with you, I'll make the change.

s0up4200 replied:
Im with gl0ryus, as I don't think a quick start guide should be focused specifically towards the arrs. And the use of "Servarr instance" would create more confusion for everybody else I think.

I get your point though!

rpatterson replied:
Ok, I'll make the change to these specific parts of the language.  But note that much else of this Quick Start is specific to this Servarr use case.  And I'm not qualified to write any other version of a quick start.

rpatterson replied:
> 6\. (qBittorrent, Sonarr, Radarr etc)

I think the note following covers that, no?

gl0ryus commented on docs/introduction.md:
```diff
@@ -85,3 +85,72 @@ Available download clients and actions
+
+8. Add a [filter action](./filters/actions).
+
+	Add an action of the matching Servarr type (e.g. Sonarr type action for a Sonarr
```
Add an action of the matching download client.

rpatterson replied:
Ditto

gl0ryus commented on docs/introduction.md:
```diff
@@ -85,3 +85,72 @@ Available download clients and actions
+3. [Register a nick on your indexer's IRC
+   network](https://autobrr.com/configuration/irc#registering-with-nickserv).
+
+4. [Group a "bot" nick with your real nick](./configuration/irc#grouping-nicks)
```
A note to check your trackers IRC guide so they don't forget any invite commands or grouping nomenclature (|bot or |autodl)

gl0ryus commented on docs/introduction.md:
```diff
@@ -85,3 +85,72 @@ Available download clients and actions
+
+2. Create a user for yourself.
+
+	Most easily done through the initial "wizzard" in the web UI when autobrr is first
```
Wizard is one z :)

rpatterson commented:
Shame, I still think this would help the newb coming in, even as is, and I don't think it would make things worse for those who already understand this domain.

zze0s commented:
> Shame, I still think this would help the newb coming in, even as is, and I don't think it would make things worse for those who already understand this domain.

It hasn't been dismissed. There was some comments on how to improve it and be a bit more generic, since not everyone is **"servarr-first"**.

rpatterson commented:
> It hasn't been dismissed. There was some comments on how to improve it and be a bit more generic, since not everyone is **"servarr-first"**.

AFAICT, I've either addressed all feedback or I'm waiting for responses.  LMK what I've missed.

s0up4200 commented:
All the links in your write-up leads to empty pages because of the `.` in your links. This happens when viewing the page with a trailing slash. You can see it for yourself here: https://deploy-preview-52--autobrr.netlify.app/introduction/

Simply remove the `.` in every link to solve that. It's not needed with how we have Docusaurus set up.

A part from that, I think the write-up is really good, but I feel it's more focused towards a Servarr setup than we want it to be (the word Servarr is used 6 times through the write-up right now). A lot of people use autobrr directly with a torrent client and I think the quick start should be as neutral as it can be towards download clients.

Also, I'd like the list to either have punctuation endings or not. Right now it's a mix of both and it makes it look a bit rushed. My preference is to not use punctuation in lists like these. The rest of the docs should (hopefully) point that out.

Please don't take any of this as bad criticism. The write-up is very nice, but I want it to be more neutral towards download clients than it is right now.

s0up4200 commented:
@rpatterson Reading my last comment I feel like it may look like I come off as an ass. I'm sorry about that. Bottom line I think your write-up is very good.

rpatterson commented:
> Bottom line I think your write-up is very good.

That is nice to hear and clarifies the tone, thanks!

> All the links in your write-up leads to empty pages because of the . in your links. This happens when viewing the page with a trailing slash. You can see it for yourself here: https://deploy-preview-52--autobrr.netlify.app/introduction/

From my previous reply, [your docs have existing relative links from before my changes](https://github.com/autobrr/autobrr.com/pull/52#issuecomment-1353162072).  Also, `./...` relative links in the Markdown source are rendered as absolute `href="/..."` URLs in the HTML the same as relative links without the leading `./`.  IOW, the rendered HTML has the correct links with or without the the trailing `/`:

```
$ curl -v https://deploy-preview-52--autobrr.netlify.app/introduction/#quick-start
...
Add an <a href="/configuration/indexers">indexer</a>
...
```

I'm guessing there's some JS intercepting the click event that transforms those correct URLs in the HTML into requests for incorrect URLs in the browser, in which case it's behavior is definitely broken.  Also, the link to the introduction from [the site root](https://deploy-preview-52--autobrr.netlify.app/) doesn't have a trailing `/` in both the rendered HTML and in the resulting URL in the browser when clicked but the HTTP server responds to direct requests for the URL without a trailing `/` with a `301` redirect to the URL *with* a trailing `/`.  This combination leads to behavior whose brokenness is greater than the sum of it's broken parts:

1. Susan visits the site root in a browser
2. Susan clicks on the "Introduction" and then "autobrr" under "Configuration" in the left nav (IOW, *not* my changes)
3. Susan clicks the "Indexers" link under "Other options" and sees the correct indexers docs
4. Susan clicks her browser's back button, copies the browser URL and pastes it to Bob
5. Bob visits the pasted URL in his browser
6. Bob clicks the "Indexers" link under "Other options" and sees the "Page Not Found" page

But as you say, this is *not* the behavior when the Markdown source uses an absolute link even though *both forms* of relative links *and* absolute links are *all* rendered as absolute links in the rendered HTML.  Not a good showing from Docusaurus.

At any rate, I've converted all relative links in my changes to absolute links.

rpatterson commented:
> Also, I'd like the list to either have punctuation endings or not. Right now it's a mix of both and it makes it look a bit rushed. My preference is to not use punctuation in lists like these. The rest of the docs should (hopefully) point that out.

Ironic that you think it looks rushed given how much time this has taken.  ;-)  At any rate, changed and pushed.

> A part from that, I think the write-up is really good, but I feel it's more focused towards a Servarr setup than we want it to be (the word Servarr is used 6 times through the write-up right now). A lot of people use autobrr directly with a torrent client and I think the quick start should be as neutral as it can be towards download clients.

See [my response](https://github.com/autobrr/autobrr.com/pull/52#discussion_r1049790773) in our previous discussion concerning this.  To repeat, I am not qualified to write a general form of this.  I think, using a specific example will still be more helpful to a user coming in with a different specific use case in terms of understanding how things fit together than the absence of any such example.  But you do you.

Regardless, I can't afford to put any more time into this.  If you don't merge this as is but want to do something with it in the future, then make a branch off of my fork so you don't lose it if I clean up my forks later.

s0up4200 commented on docs/introduction.md:
```diff
@@ -85,3 +85,77 @@ Available download clients and actions
+	started.
+
+3. [Register a nick on your indexer's IRC
+   network](configuration/irc#registering-with-nickserv).
```
This one is still broken. Change this one to `[Register a nick on your indexer's IRC
   network](/configuration/irc#registering-with-nickserv)` and I think we're good!

rpatterson replied:
Done.

s0up4200 commented:
Thanks! Merged 👍
